crates/metrics/collectors/economic/src/lib.rs

```diff
@@ -3,65 +3,191 @@
+impl Collector {
+    /// Operator stakes in AVS registry contract.
+    /// Most commonly represents a weighted combination of delegated shares in the DelegationManager EigenLayer contract.
+    pub fn new(
+        logger: SharedLogger,
+        operator_addr: Address,
+        operator_id: OperatorId,
+        el_reader: ELChainReader,
+        avs_registry_reader: AvsRegistryChainReader,
+        quorum_names: HashMap<u64, String>,
+        avs_name: &str,
+    ) -> Self {
+        describe_gauge!(
+            "eigen_registered_stakes",
+            "A gauge with weighted delegation of delegated shares in delegation manager contract"
+        );
+        // for now the namespace if "e" instead of eigen as the docs mention to not use the eigen namespace for this metric
+        describe_gauge!(
+            format!("{},_slashing_status", "e"),
+            "Whether the operator has been slashed"
+        );
 
-        gauge
+        Self {
+            logger,
+            operator_addr,
+            el_reader,
+            avs_registry_reader,
+            operator_id,
+            quorum_names,
+            avs_name: avs_name.to_owned(),
+        }
     }
 
-    pub fn set_stake(&self, quorum_number: &str, quorum_name: &str, value: f64) {
-        // Create the metric key with dynamic labels
+    pub fn set_stake(&self, quorum_number: &str, quorum_name: &str, avs_name: &str, value: f64) {
+        // Create the metric key with dynamic
         let key = Key::from_parts(
             "eigen_registered_stakes",
             vec![
                 Label::new("quorum_number", quorum_number.to_string()),
                 Label::new("quorum_name", quorum_name.to_string()),
+                Label::new("avs_name", avs_name.to_string()),
             ],
         );
+        gauge!(key.to_string()).set(value);
         self.logger.debug(
             &format!(
-                "set registered stakes , quorum_name: {} , quorum_number: {} , value: {}  ",
-                quorum_name, quorum_number, value
-            ),
+            "set registered stakes , quorum_name: {} , quorum_number: {} , avs_name: {}, value: {}",
+            quorum_name, quorum_number, avs_name, value
+        ),
             "eigen-metrics-collectors-economic.set_stake",
         );
+    }
+
+    pub async fn collect(&mut self) -> Result<(), CollectorMetricError> {
+        let operator_is_frozen = self
+            .el_reader
+            .operator_is_frozen(self.operator_addr)
+            .await?;
 
-        // Register or retrieve the gauge with the specified key and set the value
-        metrics::gauge!(key.to_string(), value);
+        let mut operator_is_frozen_float = 0.0;
+        if operator_is_frozen {
+            operator_is_frozen_float = 1.0;
+        }
+
+        gauge!("e_slashing_status").set(operator_is_frozen_float);
+
+        let operator_id_res = self.init_operator_id().await;
+
+        match operator_id_res {
+            Ok(_) => {
```

Comment:
This match could be improved doing an early return with an `if`:
and something like:
```
let operator_id_res = self.init_operator_id().await.inspect_err(|e| elf.logger.warn(&format!("Failed to fetch and cache operator id. Skipping collection of registeredStake metric. , err {}",e),"eigen-metrics-collectors-economic.collect"););
```